### ISSUE TITLE
Use the `source.sortImports` code action instead to avoid deleting "unused" imports

### ISSUE THREAD

**Issue by jaydenseric** [simonhaenisch/prettier-plugin-organize-imports#37]
The [`source.sortImports`](https://code.visualstudio.com/updates/v1_57#_sort-imports-source-action) code action is similar to the what organize imports does, except it doesn't doesn't remove "unused" imports.

Automatic removal of "unused" imports by a Prettier plugin is unacceptable, for a few reasons:

1. Automatically deleting imports that could have intended side-effects is more risky than just reordering them.

2. It destructively modifies the code you are working on. Often you begin writing a new module by writing the imports you expect to use, then start writing the module code. If you trigger a Prettier format at any point (e.g. by saving the file) before you have used all the imports in your module code, the imports unhelpfully get stripped out.

3. It destroys JS code examples in markdown readme files that demo how to import things. In these situations the imports are expected to be unused, since the JS is just an example snippet. The result after running Prettier on the markdown is all the JS fenced code blocks containing just imports are empty. Here is an example of "unused" imports within markdown:

   <img width="600" alt="Screen Shot 2021-10-17 at 2 26 17 pm" src="https://user-images.githubusercontent.com/1754873/137609731-b9df6d03-9176-46f1-ae1b-102f3cd2a110.png">
    
   I've found it difficult to reliably reproduce when the TS language server considers the imports "unused", when using Prettier with your `prettier-plugin-organize-imports` in a VS Code editor open to a markdown file with JS fenced code blocks containing single import statements, they don't get deleted on format. But, the exact same markdown when [Prettier formatted programmatically as part of the `jsdoc-md` readme generation](https://github.com/jaydenseric/jsdoc-md/blob/7df4bd2fb78101fe58449b65ebcf77697ce4f7dd/public/jsdocMd.mjs#L117-L154) does result in deleted imports.

Somewhat related:

- https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/24
- https://github.com/simonhaenisch/prettier-plugin-organize-imports/issues/25

**Comment by Eoksni** [comment]
> Another possible solution going forward is to add a boolean removeUnusedImports config option to the plugin that switches to using source.sortImports if false (true by default). I'm open to a PR for that, seems pretty straightforward.

I would totally love to see this implemented. I press ctrl-s all the time just for the sake of it (habit "from the old days") and sometimes I lose my imports unintentionally.
I'll try to find some time at the weekend to raise a PR but ... no promises.

**Comment by localpcguy** [comment]
Just a reminder that not everyone is going to use this with VSCode, so if options are added that explicitly use VSCode functionality (rather than Prettier or Typescript), that should be made very clear in the README or at least in that option. I know the README currently mentions 'This is the same as using the "Organize Imports" action in VS Code.' but that isn't quite accurate given this discussion, I believe.

**Comment by simonhaenisch** [comment]
Hey @localpcguy not sure i follow... this plugin is for Prettier and has nothing to do with VS Code per-se. It uses a method of the TypeScript language service API called `source.organizeImports`, and VS Code (when in TypeScript language mode) has an action called "Organize Imports", that calls the same method of the same API. So yeah, using the VS Code action or running Prettier with this plugin does effectively the same thing to your import statements.

Maybe you misunderstood something, or else you can clarify why the readme would be inaccurate?

There's not gonna be any options specifically for VS Code, not even sure how that would be possible... this is not an editor plugin but a Prettier plugin.

**Comment by simonhaenisch** [comment]
In case anyone wants to PR this:

- Expose a new boolean option `sortOnly` (see https://prettier.io/docs/en/plugins.html#options)

https://github.com/simonhaenisch/prettier-plugin-organize-imports/blob/5ba4618ba24e98e88a781d4fb9b7bfc00caa3cfd/lib/organize.js#L15

- change ☝️ this to sth like

```js
const method = options.sortOnly ? languageService.sortImports : languageService.organizeImports;

const fileChanges = method({ type: 'file', fileName }, {}, {})[0];
```

(assuming those two methods have the same signature)

- add some tests

**Comment by simonhaenisch** [comment]
Finally had some time to make a PR 👆 for this in case anyone is keen to have a look before I merge it.

**Comment by simonhaenisch** [comment]
Just released `3.1.0` so in case you want to use this, you can upgrade, then add `organizeImportsSkipDestructiveCodeActions: true` to your Prettier config.
